src/test/regress/expected/with_basics.out

```diff
@@ -0,0 +1,274 @@
+	WITH another_cte AS (
+		SELECT user_id FROM events_table WHERE value_2 IN (1, 2, 3)
+	)
+	SELECT * FROM another_cte
+)
+ORDER BY
+	1, 2
+LIMIT
+	10;
+ERROR:  cannot pushdown the subquery since all relations are not joined using distribution keys
+DETAIL:  Each relation should be joined with at least one another relation using distribution keys and equality operator.
+-- CTEs in FROM should work
+WITH cte AS (
+	SELECT user_id, value_2 from users_table WHERE user_id IN (1, 2) ORDER BY 2 LIMIT 5
+)
+SELECT * FROM cte;
+ user_id | value_2 
+---------+---------
+       1 |       0
+       2 |       0
+       2 |       1
+       2 |       1
+       2 |       1
+(5 rows)
+
+-- create and use CTE in FROM should work
+SELECT * FROM (WITH cte AS (
+		SELECT user_id, value_2 from users_table WHERE user_id IN (1, 2) ORDER BY 2 LIMIT 5
+	)
+SELECT * FROM cte) a;
+ERROR:  cannot push down this subquery
+DETAIL:  CTEs in multi-shard queries are currently unsupported
+--CTEs in FROM in a subquery should work
+SELECT * FROM (WITH cte AS (
+		SELECT user_id, value_2 from users_table WHERE user_id IN (1, 2) ORDER BY 2 LIMIT 5
+	)
+SELECT * FROM cte) b;
+ERROR:  cannot push down this subquery
+DETAIL:  CTEs in multi-shard queries are currently unsupported
+--CTEs in FROM in a subquery should work even when we create it there
+SELECT * FROM (
+	SELECT * FROM (WITH cte AS (
+		SELECT user_id, value_2 from users_table WHERE user_id IN (1, 2) ORDER BY 2 LIMIT 5
+		)
+		SELECT * FROM cte
+	) a) b;
+ERROR:  cannot push down this subquery
+DETAIL:  CTEs in multi-shard queries are currently unsupported
+-- CTEs in FROM and as a subquery in FROM should work
+WITH users_events AS (
+	WITH users_done_2_3 AS (
+		SELECT users_table.user_id, users_table.value_2 FROM users_table, events_table WHERE users_table.user_id = events_table.user_id AND event_type IN (2, 3)
+	)
+	SELECT users_done_2_3.user_id, users_done_2_3.value_2 as value_2_3 FROM users_done_2_3
+)
+SELECT * FROM (SELECT * FROM users_events WHERE value_2_3 IN (2, 5)) a ORDER BY 1, 2 LIMIT 10;
+ user_id | value_2_3 
+---------+-----------
+       1 |         2
+       1 |         2
+       1 |         2
+       1 |         2
+       1 |         2
+       2 |         2
+       2 |         2
+       2 |         2
+       2 |         2
+       2 |         2
+(10 rows)
+
+-- SELECT * FROM (SELECT * FROM cte UNION SELECT * FROM distributed_table) a; should error out
+WITH cte AS (
+	SELECT * FROM users_table
+)
+SELECT * FROM (
+	SELECT * FROM cte UNION (SELECT * FROM events_table)
+	) a
+ORDER BY 
+	1,2,3,4,5,6
+LIMIT 
+	10;
+ERROR:  cannot push down this subquery
+DETAIL:  Complex subqueries and CTEs are not supported within a UNION
+SELECT * FROM (
+	SELECT * FROM (WITH cte AS (
+			SELECT * FROM users_table
+		)
+		SELECT * FROM cte
+	)b UNION (SELECT * FROM events_table)) a
+ORDER BY 
+1,2,3,4,5,6
+LIMIT 
+10;
+ERROR:  cannot push down this subquery
+DETAIL:  CTEs in multi-shard queries are currently unsupported
```

Comment:
Ah, we should maybe update this error message?  `CTEs inside the subqueries are currently unsupported`. 

Actually, we could be able to support this in subquery PR, so lets keep it for now?

```SQL
SELECT users_table.user_id
FROM users_table,

  (WITH cte AS
     (SELECT user_id
      FROM events_table) SELECT *
   FROM cte) AS foo
WHERE foo.user_id = users_table.user_id
  AND users_table.value_2 IN (8,
                              9)
```